pyodide-build/pyodide_build/common.py

```diff
@@ -1,7 +1,45 @@
 import functools
 import subprocess
 from pathlib import Path
-from typing import Optional
+from typing import Iterable, Iterator, Optional
+
+from packaging.tags import Tag, compatible_tags, cpython_tags, parse_tag
+
+PLATFORM = "emscripten_wasm32"
+
+
+def pyodide_tags() -> Iterator[Tag]:
+    """
+    Returns the sequence of tag triples for the Pyodide interpreter.
+
+    The sequence is ordered in decreasing specificity.
+    """
+
+    yield from cpython_tags(platforms=[PLATFORM])
+    yield from compatible_tags(platforms=[PLATFORM])
+
+
+def find_matching_wheels(wheel_paths: Iterable[Path]) -> Iterator[Path]:
+    """
+    Returns the sequence wheels whose tags match the Pyodide interpreter.
+
+    Parameters
+    ----------
+    wheel_paths
+        A list of paths to wheels
+
+    Returns
+    -------
+    The subset of wheel_paths that have tags that match the Pyodide interpreter.
+    """
+    wheel_tags_list: list[tuple[Path, frozenset[Tag]]] = [
+        (wheel, parse_tag(wheel.stem.split("-", 2)[-1])) for wheel in wheel_paths
```

Comment:
can't we use [parse_wheel_filename](https://packaging.pypa.io/en/latest/utils.html#packaging.utils.parse_wheel_filename) instead? (for here and other places)

Reply:
Good point.